app/src/main/java/com/g11x/checklistapp/ChecklistActivity.java

```diff
@@ -74,19 +85,19 @@ public void onClick(View view) {
     recyclerView.setAdapter(checklistAdapter);
   }
 
-  @Override
-  protected void onStart() {
-    super.onStart();
+  private void onLanguageChange(String newValue) {
+    language = Language.valueOf(newValue);
   }
 
   @Override
   protected void onDestroy() {
     super.onDestroy();
     checklistAdapter.cleanup();
+    languageChangeListener.unregister(this);
```

Comment:
Instead of creating a duplicate listener in future we should probably allow NavigationActivity have a delegate method that can be implemented (or not).